subsys/mgmt/mcumgr/smp/src/smp.c

```diff
@@ -190,15 +195,20 @@ static int smp_handle_single_payload(struct smp_streamer *cbuf, const struct smp
 	if (handler_fn) {
 		bool ok;
 
-		*handler_found = true;
-		ok = zcbor_map_start_encode(cbuf->writer->zs,
-					    CONFIG_MCUMGR_SMP_CBOR_MAX_MAIN_MAP_ENTRIES);
+#if defined(CONFIG_MCUMGR_MGMT_CUSTOM_PAYLOAD)
+		if (!group->custom_payload) {
+#endif
```

Comment:
This probably could be replaced with
```
if (!IS_ENABLED(CONFIG_MCUMGR_MGMT_CUSTOM_PAYLOAD) || !group->custom_payload) {
```
to reduce the #ifdefs

Reply:
No as that will fail compilation if the Kconfig is not enabled due to the field not existing